DigitalLearningSolutions.Web/Scripts/trackingSystem/reports.ts

```diff
@@ -34,40 +36,48 @@ function generateChart(request: XMLHttpRequest) {
     response = JSON.parse(response);
   }
   const data = constructChartistData(response);
-
-  const options = {
-    axisY: {
-      scaleMinSpace: 10,
-      onlyInteger: true,
-    },
-    chartPadding: {
-      bottom: 32,
-    },
-  };
-
-  const chart = new Chartist.Line('.ct-chart', data, options);
-
-  chart.on('draw',
-    // The type here is Chartist.ChartDrawData, but the type specification is missing getNode()
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    (drawnElement: any) => {
-      const { element } = drawnElement;
-      // IE renders text in SVGs with 'text' tags that do not work with most CSS properties
-      // so we set the relevant attributes manually
-      if (element.getNode().tagName === 'text'
-        && element.classes().indexOf('ct-horizontal') >= 0
-        && element.classes().indexOf('ct-label') >= 0) {
-        const xOrigin = Number(element.getNode().getAttribute('x'));
-        const yOrigin = element.getNode().getAttribute('y');
-        const width = Number(element.getNode().getAttribute('width'));
-        // this should match the NHS tablet breakpoint
-        const mediaQuery = window.matchMedia('(min-width: 641px)');
-        const rotation = mediaQuery.matches ? -45 : -60;
-        element.attr({
-          transform: `translate(-${width}) rotate(${rotation} ${xOrigin + width} ${yOrigin})`,
-        });
-      }
-    });
+  const numberOfEntries = data.labels?.length;
+
+  if (numberOfEntries !== undefined && numberOfEntries <= maxNumberOfEntriesForActivityGraph) {

```

Comment:
Think we could take the chance to improve the code by extracting the first branch of the if-statement into a function of its own. It'd also make it easier to see what this if-statement does.